src/radical/pilot/agent/agent_0.py

```diff
@@ -252,23 +252,6 @@ def finalize(self):
         # NOTE: we do not push the final pilot state, as that is done by the
         #       bootstrapper *after* this pilot *actually* finished.
 
-        self._log.info('pilot state: %s [%s]', state, self._final_cause)
-
-        out, out, err = None, None, None
-
-        try   : out = open('./agent.0.out', 'r').read(1024)
-        except: pass
-        try   : err = open('./agent.0.err', 'r').read(1024)
-        except: pass
-        try   : log = open('./agent.0.log', 'r').read(1024)
-        except: pass
-
-        self._dbs._c.update({'type': 'pilot',
-                             'uid' : self._pid},
-                            {'$set': {'stdout' : rpu.tail(out),
-                                      'stderr' : rpu.tail(err),
-                                      'logfile': rpu.tail(log)} })
-
```

Comment:
This code was removed as it duplicated a DB update done via the update worker.

Reply:
Are we sure that the other copy of this code is *always* called? Do we have a test for it?

Reply:
Yes - 4 lines above the remove part, `update_db()` is *always* called, and `update_db()` *always* runs the same lines (20 lines up in the same file).